### DISCUSSION TITLE
Test failures while trying to update to v0.115.0

### DISCUSSION THREAD

**Discussion by icp1994** [dathere/qsv#1327] [General]
I was not sure if this belonged in under `Issues` so started a discussion instead.

Is there some requirement changes for running tests between 114 & 115? I'm getting 5 failures

```
failures:

---- test_sniff::sniff_url_snappy stdout ----
thread 'test_sniff::sniff_url_snappy' panicked at 'assertion failed: got.ends_with(expected_end)', tests/test_sniff.rs:149:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_sniff::sniff_url_notcsv stdout ----
thread 'test_sniff::sniff_url_notcsv' panicked at 'assertion failed: got_error.starts_with(expected)', tests/test_sniff.rs:85:5

---- test_sniff::sniff_url_snappy_noinfer stdout ----
thread 'test_sniff::sniff_url_snappy_noinfer' panicked at 'assertion failed: got.starts_with(expected)', tests/test_sniff.rs:164:5

---- test_snappy::snappy_decompress_url stdout ----
thread 'test_snappy::snappy_decompress_url' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"case_enquiry_id,open_dt,target_dt,closed_dt, [...removed for brevity...]

---- test_validate::validate_adur_public_toilets_dataset_with_json_schema_url stdout ----
thread 'test_validate::validate_adur_public_toilets_dataset_with_json_schema_url' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', tests/workdir.rs:223:14
 ```
 
 Do these need some external dependency or feature flags? FWIW, I'm building `qsv` with
 ```
 --no-default-features --features=feature_capable
 ```

**Comment by jqnatividad** [comment]
Hi @icp1994,

What platform are you running the tests on? I don't have a GH actions workflow for that combination and will be sure to add it.

Regardless, I got a different error than you on macOS Sonoma running on an M2 Pro with 32gb memory.

```cargo t --no-default-features -F feature_capable ```

```
failures:

---- test_comments::envlist stdout ----
thread 'test_comments::envlist' panicked at 'assertion failed: `(left == right)`
  left: `"QSV_ENVVAR: \"#\"\nQSV_KIND: \"compiled\""`,
 right: `"MIMALLOC_ENVVAR: \"1\"\nQSV_ENVVAR: \"#\""`', tests/test_comments.rs:248:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_comments::envlist

test result: FAILED. 977 passed; 1 failed; 3 ignored; 0 measured; 0 filtered out; finished in 12.25s
```

but this one error is because I assumed `mimalloc` is used, as its a default feature.

**Comment by icp1994** [comment]
It turned out to be completely unrelated to `qsv`. Since I was running the tests in parallel and those test functions were among the ones which downloads data from GH, it was rate-limiting me.

**Reply by jqnatividad** [reply]
Aah yes.  That does happen from time to time.

It used to happen more often in the past but I noticed GH rate-limiting has eased up a bit lately.

I'll make sure to make the tests that depend on GH downloads to be more robust.

**Comment by icp1994** [comment]
For some context, I maintain the `qsv` [package](https://github.com/void-linux/void-packages/commits/master/srcpkgs/qsv) in Void Linux.

**Reply by jqnatividad** [reply]
That's awesome! Thanks for maintaining it! Hopefully, qsv's fast release tempo hasn't caused too many problems... 😄
